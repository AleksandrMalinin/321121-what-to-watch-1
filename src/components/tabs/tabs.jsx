import React, {PureComponent} from 'react';
import PropTypes from 'prop-types';
import {constants} from '../../constants.js';
import {getRatingLevel} from '../../utils.js';
import withActiveItem from '../../hocs/with-active-item/with-active-item.js';
import TabContent from '../tab-content/tab-content.jsx';

class Tabs extends PureComponent {
  constructor() {
    super();

    this.onTabSwitch = (genre) => this._onTabSwitch.bind(this, genre);
  }

  render() {
    const {movie, activeItem} = this.props;

    return <React.Fragment>
      <nav className="movie-nav movie-card__nav">
        <ul className="movie-nav__list">
          {constants.TAB_NAMES.map((tabName, i) => <li className={`movie-nav__item ` + (activeItem === tabName ? `movie-nav__item--active` : ``)} key={i}>
            <a href="#" className="movie-nav__link" onClick={this.onTabSwitch(tabName)}>{tabName}</a>
          </li>)}
        </ul>
      </nav>

      <TabContent activeItem={activeItem} tabName={`Overview`}>
        <div className="movie-rating">
          <div className="movie-rating__score">{movie ? movie.rating : ``}</div>
          <p className="movie-rating__meta">
            <span className="movie-rating__level">{movie ? getRatingLevel(movie.rating) : ``}</span>
            <span className="movie-rating__count">{movie ? movie.scores_count : ``} ratings</span>
          </p>
        </div>

        <div className="movie-card__text">
          <p>{movie ? movie.description : ``}</p>

          <p className="movie-card__director"><strong>{movie ? movie.director : ``}</strong></p>

          <p className="movie-card__starring"><strong>{movie ? movie.starring.join(`, `) : ``} and other</strong></p>
        </div>
      </TabContent>

      <TabContent activeItem={activeItem} tabName={`Details`}>
        <div className="movie-card__text movie-card__row">
          <div className="movie-card__text-col">
            <p className="movie-card__details-item">
              <strong className="movie-card__details-name">Director</strong>
              <span className="movie-card__details-value">{movie ? movie.director : ``}</span>
            </p>
            <p className="movie-card__details-item">
              <strong className="movie-card__details-name">Starring</strong>
              <span className="movie-card__details-value">
                {movie ? movie.starring.join(`, `) : ``}
              </span>
            </p>
          </div>

          <div className="movie-card__text-col">
            <p className="movie-card__details-item">
              <strong className="movie-card__details-name">Run Time</strong>
              <span className="movie-card__details-value">{movie ? movie.running : ``}</span>
            </p>
            <p className="movie-card__details-item">
              <strong className="movie-card__details-name">Genre</strong>
              <span className="movie-card__details-value">{movie ? movie.genre : ``}</span>
            </p>
            <p className="movie-card__details-item">
              <strong className="movie-card__details-name">Released</strong>
              <span className="movie-card__details-value">{movie ? movie.released : ``}</span>
            </p>
          </div>
        </div>
      </TabContent>

      <TabContent activeItem={activeItem} tabName={`Reviews`}>
        <div className="movie-card__reviews movie-card__row">
          <div className="movie-card__reviews-col">
            <div className="review">
              <blockquote className="review__quote">
                <p className="review__text">Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director&rsquo;s funniest and most exquisitely designed movies in years.</p>

                <footer className="review__details">
                  <cite className="review__author">Kate Muir</cite>
                  <time className="review__date" dateTime="2016-12-24">December 24, 2016</time>
                </footer>
              </blockquote>

              <div className="review__rating">8,9</div>
            </div>

            <div className="review">
              <blockquote className="review__quote">
                <p className="review__text">Anderson&rsquo;s films are too precious for some, but for those of us willing to lose ourselves in them, they&rsquo;re a delight. &quot;The Grand Budapest Hotel&quot; is no different, except that he has added a hint of gravitas to the mix, improving the recipe.</p>

                <footer className="review__details">
                  <cite className="review__author">Bill Goodykoontz</cite>
                  <time className="review__date" dateTime="2015-11-18">November 18, 2015</time>
                </footer>
              </blockquote>

              <div className="review__rating">8,0</div>
            </div>

            <div className="review">
              <blockquote className="review__quote">
                <p className="review__text">I didn&rsquo;t find it amusing, and while I can appreciate the creativity, it&rsquo;s an hour and 40 minutes I wish I could take back.</p>

                <footer className="review__details">
                  <cite className="review__author">Amanda Greever</cite>
                  <time className="review__date" dateTime="2015-11-18">November 18, 2015</time>
                </footer>
              </blockquote>

              <div className="review__rating">8,0</div>
            </div>
          </div>
          <div className="movie-card__reviews-col">
            <div className="review">
              <blockquote className="review__quote">
                <p className="review__text">The mannered, madcap proceedings are often delightful, occasionally silly, and here and there, gruesome and/or heartbreaking.</p>

                <footer className="review__details">
                  <cite className="review__author">Matthew Lickona</cite>
                  <time className="review__date" dateTime="2016-12-20">December 20, 2016</time>
                </footer>
              </blockquote>

              <div className="review__rating">7,2</div>
            </div>

            <div className="review">
              <blockquote className="review__quote">
                <p className="review__text">It is certainly a magical and childlike way of storytelling, even if the content is a little more adult.</p>

                <footer className="review__details">
                  <cite className="review__author">Paula Fleri-Soler</cite>
                  <time className="review__date" dateTime="2016-12-20">December 20, 2016</time>
                </footer>
              </blockquote>

              <div className="review__rating">7,6</div>
            </div>

            <div className="review">
              <blockquote className="review__quote">
                <p className="review__text">It is certainly a magical and childlike way of storytelling, even if the content is a little more adult.</p>

                <footer className="review__details">
                  <cite className="review__author">Paula Fleri-Soler</cite>
                  <time className="review__date" dateTime="2016-12-20">December 20, 2016</time>
                </footer>
              </blockquote>

              <div className="review__rating">7,0</div>
            </div>
          </div>
        </div>
      </TabContent>
    </React.Fragment>;
  }

  _onTabSwitch(name, event) {
    event.preventDefault();
    this.props.onChange(name);
  }
}

Tabs.propTypes = {
  movie: PropTypes.objectOf(PropTypes.shape({
    name: PropTypes.string,
    posterImage: PropTypes.string,
    previewImage: PropTypes.string,
    genre: PropTypes.string
  })),
  activeItem: PropTypes.string,
  onChange: PropTypes.func
};

export default withActiveItem(constants.DEFAULT_TAB)(Tabs);
