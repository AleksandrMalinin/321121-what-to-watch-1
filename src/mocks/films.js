/* eslint camelcase: ["error", {properties: "never"}] */

export const films = [
  {
    title: `Fantastic Beasts: The Crimes of Grindelwald`,
    preview_image: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    preview_video_link: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    genre: `Comedies`
  },
  {
    title: `Bohemian Rhapsody`,
    preview_image: `img/bohemian-rhapsody.jpg`,
    preview_video_link: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    genre: `Crime`
  },
  {
    title: `Macbeth`,
    preview_image: `img/macbeth.jpg`,
    preview_video_link: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    genre: `Documentary`
  },
  {
    title: `Aviator`,
    preview_image: `img/aviator.jpg`,
    preview_video_link: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    genre: `Dramas`
  },
  {
    title: `We need to talk about Kevin`,
    preview_image: `img/we-need-to-talk-about-kevin.jpg`,
    preview_video_link: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    genre: `Horror`
  }
];
