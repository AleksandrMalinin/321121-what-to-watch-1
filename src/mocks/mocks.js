/* eslint camelcase: ["error", {properties: "never"}] */

export const films = [
  {
    name: `Fantastic Beasts: The Crimes of Grindelwald`,
    preview_image: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    preview_video_link: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    genre: `Comedies`
  },
  {
    name: `Bohemian Rhapsody`,
    preview_image: `img/bohemian-rhapsody.jpg`,
    preview_video_link: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    genre: `Crime`
  },
  {
    name: `Macbeth`,
    preview_image: `img/macbeth.jpg`,
    preview_video_link: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    genre: `Documentary`
  },
  {
    name: `Aviator`,
    preview_image: `img/aviator.jpg`,
    preview_video_link: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    genre: `Dramas`
  },
  {
    name: `We need to talk about Kevin`,
    preview_image: `img/we-need-to-talk-about-kevin.jpg`,
    preview_video_link: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    genre: `Horror`
  }
];

export const defaultMovie = {
  background_color: `#AD9F8B`,
  background_image: `https://es31-server.appspot.com/wtw/static/film/background/Dardjeeling_Limited.jpg`,
  description: `A year after their father's funeral, three brothers travel across India by train in an attempt to bond with each other.`,
  director: `Wes Anderson`,
  genre: `Adventure`,
  id: 1,
  is_favorite: false,
  name: `Dardjeeling Limited`,
  poster_image: `https://es31-server.appspot.com/wtw/static/film/poster/Dardjeeling_Limited.jpg`,
  preview_image: `https://es31-server.appspot.com/wtw/static/film/preview/dardjeeling_limited.jpg`,
  preview_video_link: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  rating: 3.6,
  released: 2007,
  run_time: 91,
  scores_count: 165106,
  starring: [`Owen Wilson`, `Adrien Brody`, `Jason Schwartzman`],
  video_link: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`
};
