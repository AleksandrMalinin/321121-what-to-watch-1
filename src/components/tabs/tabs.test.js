import React from 'react';
import renderer from 'react-test-renderer';
import {Tabs} from './tabs.jsx';
import {defaultMovie} from '../../mocks/mocks.js';

it(`Tabs correctly renders`, () => {
  const mockFunction = jest.fn();
  const tree = renderer
  .create(
      <Tabs
        movie={defaultMovie}
        onChange={mockFunction}
        onLoadComments={mockFunction}
      />
  )
  .toJSON();

  expect(tree).toMatchSnapshot();
});
