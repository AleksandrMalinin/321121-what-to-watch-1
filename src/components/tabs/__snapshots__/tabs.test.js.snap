// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs correctly renders 1`] = `
Array [
  <nav
    className="movie-nav movie-card__nav"
  >
    <ul
      className="movie-nav__list"
    >
      <li
        className="movie-nav__item movie-nav__item--active"
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Overview
        </a>
      </li>
      <li
        className="movie-nav__item "
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Details
        </a>
      </li>
      <li
        className="movie-nav__item "
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Reviews
        </a>
      </li>
    </ul>
  </nav>,
  <div>
    <div
      className="movie-rating"
    >
      <div
        className="movie-rating__score"
      >
        3.6
      </div>
      <p
        className="movie-rating__meta"
      >
        <span
          className="movie-rating__level"
        >
          normal
        </span>
        <span
          className="movie-rating__count"
        >
          165106
           ratings
        </span>
      </p>
    </div>
    <div
      className="movie-card__text"
    >
      <p>
        A year after their father's funeral, three brothers travel across India by train in an attempt to bond with each other.
      </p>
      <p
        className="movie-card__director"
      >
        <strong>
          Wes Anderson
        </strong>
      </p>
      <p
        className="movie-card__starring"
      >
        <strong>
          Owen Wilson, Adrien Brody, Jason Schwartzman
           and other
        </strong>
      </p>
    </div>
  </div>,
  <div />,
  <div />,
]
`;
